packages/console/src/components/FileUploader/index.tsx

```diff
@@ -0,0 +1,123 @@
+import type { UserAssetsResponse, AllowedUploadMimeType } from '@logto/schemas';
+import { maxUploadFileSize } from '@logto/schemas';
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { useDropzone } from 'react-dropzone';
+import { useTranslation } from 'react-i18next';
+
+import UploaderIcon from '@/assets/images/upload.svg';
+import useApi from '@/hooks/use-api';
+
+import { Ring } from '../Spinner';
+import * as styles from './index.module.scss';
+
+const allowedFileCount = 1;
+
+type Props = {
+  maxSize: number; // In kilobytes
+  allowedMimeTypes: AllowedUploadMimeType[];
+  limitDescription: string;
+  onCompleted: (fileUrl: string) => void;
+};
+
+const FileUploader = ({ maxSize, allowedMimeTypes, limitDescription, onCompleted }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [isUploading, setIsUploading] = useState(false);
+  const [uploaderError, setUploaderError] = useState<string>();
+  const hasError = Boolean(uploaderError);
+
+  const api = useApi();
+
+  const onDrop = useCallback(
+    async (acceptedFiles: File[]) => {
+      setUploaderError(undefined);
+
+      if (acceptedFiles.length > allowedFileCount) {
+        setUploaderError(t('components.uploader.error_file_count', { count: allowedFileCount }));
+
+        return;
+      }
+
+      const selectedFile = acceptedFiles[0];
+
+      if (!selectedFile) {
+        return;
+      }
+
+      if (!allowedMimeTypes.map(String).includes(selectedFile.type)) {
+        const supportedFileTypes = allowedMimeTypes.map((type) =>
+          type.split('/')[1]?.toUpperCase()
+        );
+        setUploaderError(t('components.uploader.error_file_type', { types: supportedFileTypes }));
+
+        return;
+      }
+
+      const maxSizeInBytes = maxSize * 1024;
```

Comment:
Do we have a reference for the unit we use for the file size related props?  e.g. better align the unit of the input props and `maxUploadFileSize`.

Reply:
Updated the maxSize unit from `KB` to `Bytes`, and the `maxSize * 1024` can be removed.